### PR TITLE
[refactor] [metal] Rename some variables from args to ctx

### PR DESCRIPTION
**Pull request by k-ye**
Mostly just renaming `args` to `ctx`

Related issue = #909
WANT_LGTM=any

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: taichi/backends/metal/kernel_util.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ std::string KernelAttributes::buffers_name(Buffers b) {
   const static std::unordered_map<Buffers, std::string> m = {
       REGISTER_NAME(Root),
       REGISTER_NAME(GlobalTmps),
-      REGISTER_NAME(Args),
+      REGISTER_NAME(Context),
       REGISTER_NAME(Runtime),
```
Comment (archibate) [inline taichi/backends/metal/kernel_util.cpp:22]: Thanks, a design very similar to current OpenGL backend! May do `_args_i32_` -> `_cntx_i32` refactor iapr.

File: taichi/backends/metal/codegen_metal.cpp [hunk 2]
```diff
@@ -28,9 +28,9 @@ constexpr char kKernelThreadIdName[] = "utid_";        // 'u' for unsigned
 constexpr char kKernelGridSizeName[] = "ugrid_size_";  // 'u' for unsigned
 constexpr char kRootBufferName[] = "root_addr";
 constexpr char kGlobalTmpsBufferName[] = "global_tmps_addr";
-constexpr char kArgsBufferName[] = "args_addr";
+constexpr char kContexBufferName[] = "ctx_addr";
```
Comment (archibate) [inline taichi/backends/metal/codegen_metal.cpp:31]: typo?
```suggestion
constexpr char kContextBufferName[] = "ctx_addr";
```

Comment (k-ye) [reply taichi/backends/metal/codegen_metal.cpp:31]: Nice catch thanks!

File: taichi/backends/metal/kernel_util.cpp (hunk 1 above)
Comment (k-ye) [reply taichi/backends/metal/kernel_util.cpp:22]: Actually there is no `_args_i32_` on the Metal backend. IIRC, OpenGL had to do this due to the lack of pointer castings? But Metal does have pointer types..

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ assignees: ''
 
 ---
 
+<!-- We've collected some common issue solutions in https://taichi.readthedocs.io/en/stable/install.html#troubleshooting. Make sure you've check them out first. Hopefully they could address your problem. -->
+
```
Comment (archibate) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: Off-topic change. To make sure users have check out our doc before submit an duplicate issue.

**Pull request merged**